### DISCUSSION TITLE
Adding variety of enemy factions

### DISCUSSION THREAD

**Discussion by Breech99** [Vdauphin/HeartsAndMinds#1125] [Question and answer] [answered]
I could no longer add RHS enemy AI vehicles and infantry sprinkled into with the indy forces.. It used to work when I added them to factions lines ----- sample of mission sqf

--What am I doing wrong?????  Attempting to add in some specific RHS russian forces into the guerilla forces... It used to work and add in a mix of the factions active and the specific vehicles and infantry I selected.  Now only one factions appears in mission.

_p_en = _allfaction select _p_en; //Select faction selected from mission parameter
_allclasse = [[_p_en, "IND_F","IND_C_F","IND_G_F","OPF_F","OPF_G_F","RHS_FACTION_MSV","RHS_FACTION_RVA","RHS_FACTION_TV","RHS_FACTION_VDV","RHS_FACTION_VMF","RHS_FACTION_VPVO","RHS_FACTION_VV","RHS_FACTION_VVS","RHS_FACTION_VVS_C"], _p_en_AA, _p_en_tank] call btc_fnc_mil_class; //Create classes from factions, you can combine factions like that: [[_p_en , "IND_F"], _p_en_AA, _p_en_tank] call btc_fnc_mil_class;

//Save class name to global variable
btc_enemy_side = _allclasse select 0;
btc_type_units = _allclasse select 1;
btc_type_divers = _allclasse select 2;
btc_type_crewmen = _allclasse select 3;
btc_type_boats = _allclasse select 4;
btc_type_motorized = _allclasse select 5;
btc_type_motorized_armed = _allclasse select 6;
btc_type_mg = _allclasse select 7;
btc_type_gl = _allclasse select 8;

//Sometimes you need to remove units: - ["Blabla","moreBlabla"];
//Sometimes you need to add units: + ["Blabla","moreBlabla"];
switch (_p_en) do {
/case "Myfactionexemple" : {
btc_type_units = btc_type_units - ["Blabla","moreBlabla"];
btc_type_divers = btc_type_divers + ["Blabla","moreBlabla"];
btc_type_crewmen = "Blabla";
btc_type_boats = btc_type_boats;
btc_type_motorized = btc_type_motorized;
btc_type_mg = btc_type_mg;
btc_type_gl = btc_type_gl;
};/
case "OPF_G_F" : {
btc_type_motorized = btc_type_motorized + ["I_Truck_02_transport_F", "I_Truck_02_covered_F", "rhs_tigr_sts_3camo_vdv", "rhs_btr70_vdv", "RHS_Ural_Open_Flat_MSV_01"];
btc_type_motorized_armed = btc_type_motorized_armed + ["I_Heli_light_03_F", "rhs_tigr_sts_3camo_vdv", "RHS_Mi24P_vdv"];
};
case "IND_C_F" : {
btc_type_motorized = btc_type_motorized + ["I_G_Offroad_01_repair_F", "rhs_tigr_sts_3camo_vdv","I_G_Offroad_01_F","rhs_btr70_vdv","I_Heli_light_03_F","RHS_Ural_Open_Flat_MSV_01","I_G_Quadbike_01_F","I_C_Offroad_092_AT_F","I_C_Offroad_02_LMG_F","I_G_Van_01_fuel_F","I_Truck_02_transport_F","I_Truck_02_covered_F","rhs_tigr_sts_3camo_vdv","rhs_btr70_vdv","RHS_Ural_Open_Flat_MSV_01"];
btc_type_motorized_armed = btc_type_motorized_armed + ["I_Heli_light_03_F", "RHS_Mi24P_vdv", "I_G_Offroad_01_F"];
btc_type_units = btc_type_units - ["I_C_Soldier_Camo_F"];
};
};

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> What am I doing wrong?????

I don't know, you should see errors in your .rpt 

Cheers

**Reply by Breech99** [reply]
nods.  no errors. but something was an issue. nothing in logs.. the latest version seems to work correctly again with faction rezzing.
